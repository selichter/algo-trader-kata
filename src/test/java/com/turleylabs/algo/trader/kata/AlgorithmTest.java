package com.turleylabs.algo.trader.kata;

import org.approvaltests.Approvals;
import org.approvaltests.reporters.UseReporter;
import org.approvaltests.reporters.intellij.IntelliJReporter;
import org.junit.Test;

import java.time.LocalDate;

@UseReporter(IntelliJReporter.class)
public class AlgorithmTest {

    private LocalDate startDate = LocalDate.of(2010, 3, 23);
    private LocalDate endDate = LocalDate.of(2020, 3, 6);

    @Test
    public void algorithmExecutesTrades() {
        RefactorMeAlgorithm refactorAlgorithm = new RefactorMeAlgorithm();

        refactorAlgorithm.run();

        Approvals.verify(refactorAlgorithm);
    }

    @Test
    public void algorithmExecutesTradesWithTestDoubles() {
        RefactorMeAlgorithm refactorAlgorithm = new RefactorMeAlgorithm() {
            @Override
            public void initialize() {
                super.initialize();
                this.setStartDate(startDate.getYear(), startDate.getMonthValue(), startDate.getDayOfMonth());
                this.setEndDate(endDate.getYear(), endDate.getMonthValue(), endDate.getDayOfMonth());
            }

            @Override
            public void run() {
                datesUntil(startDate, endDate).forEach(date -> processData(new FakeSlice(date)));
            }
        };

        refactorAlgorithm.run();

        Approvals.verify(refactorAlgorithm);
    }


}
