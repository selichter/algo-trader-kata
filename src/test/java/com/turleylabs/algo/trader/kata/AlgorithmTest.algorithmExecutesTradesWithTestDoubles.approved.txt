Portfolio{
}
Trades{
[Date=2011-01-04, Ticker=TQQQ, Quantity=31182, Price=3.206967]
[Date=2011-01-14, Ticker=TQQQ, Quantity=31182, Price=3.514737]
[Date=2011-01-25, Ticker=TQQQ, Quantity=32067, Price=3.417689]
[Date=2011-02-16, Ticker=TQQQ, Quantity=32067, Price=3.848069]
[Date=2011-04-07, Ticker=TQQQ, Quantity=35293, Price=3.496243]
[Date=2011-05-25, Ticker=TQQQ, Quantity=35293, Price=3.391505]
[Date=2011-07-11, Ticker=TQQQ, Quantity=33433, Price=3.580199]
[Date=2011-07-27, Ticker=TQQQ, Quantity=33433, Price=3.58477]
[Date=2012-01-20, Ticker=TQQQ, Quantity=34663, Price=3.457589]
[Date=2012-03-13, Ticker=TQQQ, Quantity=34663, Price=4.676203]
[Date=2012-06-18, Ticker=TQQQ, Quantity=39843, Price=4.068143]
[Date=2012-10-11, Ticker=TQQQ, Quantity=39843, Price=4.626744]
[Date=2012-11-30, Ticker=TQQQ, Quantity=42009, Price=4.388175]
[Date=2012-12-28, Ticker=TQQQ, Quantity=42009, Price=4.039881]
[Date=2013-01-03, Ticker=TQQQ, Quantity=36602, Price=4.636719]
[Date=2013-06-24, Ticker=TQQQ, Quantity=36602, Price=5.187006]
[Date=2013-07-10, Ticker=TQQQ, Quantity=31287, Price=6.068133]
[Date=2013-10-18, Ticker=TQQQ, Quantity=31287, Price=8.399794]
[Date=2013-11-13, Ticker=TQQQ, Quantity=29876, Price=8.7963]
[Date=2013-11-29, Ticker=TQQQ, Quantity=29876, Price=9.477095]
[Date=2013-12-18, Ticker=TQQQ, Quantity=29447, Price=9.615082]
[Date=2014-02-03, Ticker=TQQQ, Quantity=29447, Price=9.020738]
[Date=2014-02-19, Ticker=TQQQ, Quantity=24654, Price=10.774676]
[Date=2014-04-01, Ticker=TQQQ, Quantity=24654, Price=10.769689]
[Date=2014-05-12, Ticker=TQQQ, Quantity=25863, Price=10.265952]
[Date=2014-10-10, Ticker=TQQQ, Quantity=25863, Price=12.472508]
[Date=2014-11-04, Ticker=TQQQ, Quantity=21050, Price=15.324175]
[Date=2014-12-16, Ticker=TQQQ, Quantity=21050, Price=14.614169]
[Date=2015-02-04, Ticker=TQQQ, Quantity=19400, Price=15.857281]
[Date=2015-07-10, Ticker=TQQQ, Quantity=19400, Price=17.901304]
[Date=2015-07-22, Ticker=TQQQ, Quantity=16992, Price=20.437723]
[Date=2015-08-20, Ticker=TQQQ, Quantity=16992, Price=17.460548]
[Date=2015-10-13, Ticker=TQQQ, Quantity=18082, Price=16.407728]
[Date=2015-12-11, Ticker=TQQQ, Quantity=18082, Price=18.391954]
[Date=2016-03-03, Ticker=TQQQ, Quantity=21657, Price=15.356567]
[Date=2016-05-04, Ticker=TQQQ, Quantity=21657, Price=15.085462]
[Date=2016-06-01, Ticker=TQQQ, Quantity=18789, Price=17.387367]
[Date=2016-06-23, Ticker=TQQQ, Quantity=18789, Price=16.732054]
[Date=2016-07-13, Ticker=TQQQ, Quantity=17772, Price=17.690075]
[Date=2016-11-03, Ticker=TQQQ, Quantity=17772, Price=18.900904]
[Date=2016-11-25, Ticker=TQQQ, Quantity=15821, Price=21.231083]
[Date=2017-06-02, Ticker=TQQQ, Quantity=15821, Price=36.980171]
[Date=2017-07-18, Ticker=TQQQ, Quantity=16018, Price=36.524448]
[Date=2018-01-12, Ticker=TQQQ, Quantity=16018, Price=54.151321]
[Date=2018-03-06, Ticker=TQQQ, Quantity=15357, Price=56.483158]
[Date=2018-03-12, Ticker=TQQQ, Quantity=15357, Price=62.031685]
[Date=2018-05-14, Ticker=TQQQ, Quantity=17123, Price=55.631588]
[Date=2018-07-05, Ticker=TQQQ, Quantity=17123, Price=58.329342]
[Date=2018-07-16, Ticker=TQQQ, Quantity=15441, Price=64.686203]
[Date=2018-10-10, Ticker=TQQQ, Quantity=15441, Price=55.654877]
[Date=2019-01-17, Ticker=TQQQ, Quantity=19674, Price=43.678699]
[Date=2019-05-15, Ticker=TQQQ, Quantity=19674, Price=58.86871]
[Date=2019-06-19, Ticker=TQQQ, Quantity=18705, Price=61.917606]
[Date=2019-08-05, Ticker=TQQQ, Quantity=18705, Price=55.220032]
[Date=2019-09-04, Ticker=TQQQ, Quantity=16884, Price=61.177876]
[Date=2019-10-03, Ticker=TQQQ, Quantity=16884, Price=58.918694]
[Date=2019-10-17, Ticker=TQQQ, Quantity=15105, Price=65.856178]
[Date=2020-01-09, Ticker=TQQQ, Quantity=15105, Price=94.330002]
[Date=2020-01-30, Ticker=TQQQ, Quantity=14468, Price=98.480003]
[Date=2020-02-04, Ticker=TQQQ, Quantity=14468, Price=104.629997]
}
